projects/ngx-skeleton-loader/src/lib/ngx-skeleton-loader.component.ts

```diff
@@ -24,24 +27,41 @@ export class NgxSkeletonLoaderComponent implements OnInit, AfterViewInit, OnDest
   static ngAcceptInputType_animation: boolean | string;
 
   @Input()
-  count = 1;
+  count: NgxSkeletonLoaderConfig['count'];
```

Comment:
Taking the fact that all properties declared `NgxSkeletonLoaderConfig` are optional, this and all other properties declared like this will be optional as well. For example, this `count`, with this type, is `number | undefined` and I don't think this is intended, right? What about using `NonNullable` so you can keep the type as before?

[**Repro**](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcByBzEBlA1sAG2BggDsAZCAQwBNgoBhMgMwEsMAVAC2AFtUAvHADeAWABQcOAHppcbqwDOcJXCjAAjgFdW6mnEWtSAY1SkM2JERXKth83AAGVUokdwqygO6ECEqbJwALLQqHQwVKwEykZwXDAwYIoAXLIuGFoEVFAAdKwQ0lRgrNLGELy8ZNKYlohE-jJyMIoERjDJNEpUAEZEALSkoDB9rYPJcKQQfS6IDQDauOOKMFBGGAC64zMA3BIAvnAAPhOZBLviEqCQsCqkMPTMVKboWHiExGSUtPRMpGwYIgaRRQ2RcpgA-OMAOTGXTGIhQo5wKGjYCI45QqHnKQuVi8KgwfKkSHIsBQCAYdSKRTo0nkynAal9GjZXC0qFgTKKNFIqGPaI84787nYhA8fgkzA4fBEEgUah0RgsdjcPjAUUEBVrDhDEnLVbmUVlLR3EmkLS8br0c57CQSeGeZRSt6yz4Kn7lSCDO6AyRwY13cbOmUfeXfJV-dhzGEQE0wKHrbaNE6W+hIk10NiDGgNAMwABMQbIaFOPSIAB5g+85V9Fb9-tG8wmAHxJwLm1NQO1+sqkfVaYwkKAACl7-0lrxDNfdEf+AEpfVIpDAuEocnn83AhGP2OD17GfeDwXAAIyi23iPZAA)

```suggestion
  count: NonNullable<NgxSkeletonLoaderConfig['count']>;
```

Reply:
The default value will be applied whether you actually don't pass a count property or set it to null/undefined as the `forRoot` parameter. I don't think this is needed. With this, you'd force to add a count property to the method's parameter object while we are trying to override certain default values.

Example: I want to use all the default values of the library except for the appearance: I want it to be set as 'circle' by default.
With the current implementation: 
```typescript
...
import { NgxSkeletonLoaderModule } from 'ngx-skeleton-loader';
...

@NgModule({
  declarations: [
    YourAppComponent
  ],
  imports: [
    ...
    NgxSkeletonLoaderModule.forRoot({ appearance: 'circle' }),
    ...
  ],
  providers: [],
  bootstrap: [YourAppComponent]
})

export class YourAppComponent {}
```
With your changes, this implementation would throw an error because the count property is not set and you'd be force to write this:
```typescript
...
import { NgxSkeletonLoaderModule } from 'ngx-skeleton-loader';
...

@NgModule({
  declarations: [
    YourAppComponent
  ],
  imports: [
    ...
    NgxSkeletonLoaderModule.forRoot({ appearance: 'circle', count: 1 }),
    ...
  ],
  providers: [],
  bootstrap: [YourAppComponent]
})

export class YourAppComponent {}
```

You also suggested to make all properties non-optional, but that would have the same effect explained hereabove: you'de be force to overwrite all default values only to set one specific differently from the default configuration.

What I am looking to achieve is to allow the user to change any configuration item without having to change all the others (or actually set them with a specific value).

Reply:
Hey @HunteRoi, did you have a chance to open the link I posted above with repro? Let me try to explain it more techinical with this example.

When you say that `count` refers to `NgxSkeletonLoaderConfig['count']`

```ts
count: NgxSkeletonLoaderConfig['count'];
```

TS will understand it like this (because of course `NgxSkeletonLoaderConfig['count'] is optional`):

```ts
count: number | undefined;
```

which one differs from the **current** type, which is:


```ts
count: number;
```

And that's what I'm saying, now **all** properties that are declared like this will be treated as optional as well, causing a unintended effect and a possible breaking change, as for example, if you try to access `NgxSkeletonComponent.count` it could be `undefined` now.

So, that's why I suggested two approaches you can take to get rid of this unexpected type:

1. Declare it with the `NonNullable`:

```ts
count: NonNullable<NgxSkeletonLoaderConfig['count']>;
animation: NonNullable<NgxSkeletonLoaderConfig['animation']>;
// ...
```

2. Make all properties in `NgxSkeletonLoaderConfig` required and then use `Partial<NgxSkeletonLoaderConfig>` it in `NgxSkeletonLoaderModule.forRoot()` to:


```ts
static forRoot(config?: Partial<NgxSkeletonLoaderConfig>): ModuleWithProviders<NgxSkeletonLoaderModule> {}
```

<hr>

Let me know if it's clear now.

Reply:
The second option @dev054 added here looks neat. So we can have an elegant approach by using typescript support. Also, have `Partial` changing the defaults of `NgxSkeletonLoaderConfig` in a single place removed friction to maintain it

Reply:
Agree, that's why I have pushed a change with this!